core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1256,11 +1257,13 @@ class DAGScheduler(
           if (disallowStageRetryForTest) {
             abortStage(failedStage, "Fetch failure will not retry stage due to testing config",
               None)
+            abortedStage = true
           } else if (failedStage.failedOnFetchAndShouldAbort(task.stageAttemptId)) {
             abortStage(failedStage, s"$failedStage (${failedStage.name}) " +
               s"has failed the maximum allowable number of " +
               s"times: ${Stage.MAX_CONSECUTIVE_FETCH_FAILURES}. " +
               s"Most recent failure reason: ${failureMessage}", None)
+            abortedStage = true
           } else if (failedStages.isEmpty) {
```

Comment:
Instead of having the abortedStage variable, how about re-writing the "else if" statement to be:

else {
  if (failedStages.isEmpty) {
    ... stuff currently in else-if ...
  }
  failedStages += failedStage
  failedStages += mapStage
}

That eliminates the confusion of multiple abortStage variables, as @zsxwing pointed out, and also makes the relationship between (i) adding the stage to failed stages and (ii) scheduling the Resubmit event more clear.